Fix sonos tests broken by Python 3.14.3 asyncio changes

### Description

The following `sonos` tests are flaky/failing due to the Python 3.14.3 asyncio executor scheduling change (see home-assistant/core#162263 for background).

CPython PR python/cpython#142358 changed how `run_in_executor` delivers results back to the event loop. Results are now set directly on the asyncio future instead of going through `call_soon_threadsafe`, which means `await` returns immediately but other `call_soon_threadsafe` callbacks may not have run yet.

#### Affected tests

**`tests/components/sonos/test_media_player.py::test_entity_basic`**

```
tests/components/sonos/test_media_player.py:152: in test_entity_basic
    assert state == snapshot(name=f"{entity_entry.entity_id}-state")
AssertionError: assert [+ received] == [- snapshot]
    StateSnapshot({
      ...
        'group_members': list([
  -       'media_player.zone_a',
  +       None,
        ]),
      ...
    })
```


**`tests/components/sonos/test_speaker.py::test_zgs_event_group_speakers`**

```
tests/components/sonos/test_speaker.py:116: in test_zgs_event_group_speakers
    assert state.attributes["group_members"] == ["media_player.living_room"]
AssertionError: assert [None] == ['media_player.living_room']

  At index 0 diff: None != 'media_player.living_room'
```


**`tests/components/sonos/test_repairs.py::test_subscription_repair_issues`**

```
tests/components/sonos/test_repairs.py:51: in test_subscription_repair_issues
    assert not issue_registry.async_get_issue(DOMAIN, SUB_FAIL_ISSUE_ID)
AssertionError: assert not IssueEntry(active=True, breaks_in_ha_version=None,
    domain='sonos', is_fixable=False, is_persistent=False,
    issue_id='subscriptions_failed', severity=<IssueSeverity.ERROR: 'error'>,
    translation_key='subscriptions_failed',
    translation_placeholders={'device_ip': '192.168.42.2',
    'listener_address': '192.168.4.2:1400', ...})
```


#### What was done

These tests have been temporarily disabled with `@pytest.mark.skip` in PR #169046.

#### What needs to happen

The codeowner should investigate and fix the underlying timing issue in these tests. Common fixes include:
- Adding `await hass.async_block_till_done()` after service calls
- Converting sync entity methods to async
- Restructuring mocks to avoid unintended executor code paths

### Additional context

- Related PR: home-assistant/core#162263
- Disable PR: #169046

--- a/homeassistant/components/sonos/speaker.py
+++ b/homeassistant/components/sonos/speaker.py
@@ -940,12 +940,14 @@ class SonosSpeaker:
 
             for uid in group:
                 speaker = self.data.discovered.get(uid)
-                if speaker:
+                entity_id = (
+                    entity_registry.async_get_entity_id(MP_DOMAIN, DOMAIN, uid)
+                    if speaker
+                    else None
+                )
+                if speaker and entity_id:
                     self._group_members_missing.discard(uid)
                     sonos_group.append(speaker)
-                    entity_id = cast(
-                        str, entity_registry.async_get_entity_id(MP_DOMAIN, DOMAIN, uid)
-                    )
                     sonos_group_entities.append(entity_id)
                 else:
                     self._group_members_missing.add(uid)

--- a/tests/components/sonos/test_media_player.py
+++ b/tests/components/sonos/test_media_player.py
@@ -137,9 +137,6 @@ async def test_device_registry_not_portable(
     assert reg_device.area_id == area_registry.async_get_area_by_name("Zone A").id
 
 
-@pytest.mark.skip(
-    reason="Flaky due to Python 3.14.3 asyncio changes - see home-assistant/core#162263"
-)
 async def test_entity_basic(
     hass: HomeAssistant,
     async_autosetup_sonos,

--- a/tests/components/sonos/test_repairs.py
+++ b/tests/components/sonos/test_repairs.py
@@ -2,7 +2,6 @@
 
 from unittest.mock import Mock
 
-import pytest
 from soco import SoCo
 
 from homeassistant.components.sonos.const import (
@@ -19,9 +18,6 @@ from .conftest import SonosMockEvent, SonosMockSubscribe
 from tests.common import MockConfigEntry, async_fire_time_changed
 
 
-@pytest.mark.skip(
-    reason="Flaky due to Python 3.14.3 asyncio changes - see home-assistant/core#162263"
-)
 async def test_subscription_repair_issues(
     hass: HomeAssistant,
     config_entry: MockConfigEntry,
@@ -44,13 +40,12 @@ async def test_subscription_repair_issues(
 
     # Ensure the issue still exists after reload
     assert await hass.config_entries.async_reload(config_entry.entry_id)
-    await hass.async_block_till_done()
+    await hass.async_block_till_done(wait_background_tasks=True)
     assert issue_registry.async_get_issue(DOMAIN, SUB_FAIL_ISSUE_ID)
 
     # Ensure the issue has been removed after a successful subscription callback
     variables = {"ZoneGroupState": zgs_discovery}
     event = SonosMockEvent(soco, soco.zoneGroupTopology, variables)
     sub_callback(event)
-    await hass.async_block_till_done()
-    assert not issue_registry.async_get_issue(DOMAIN, SUB_FAIL_ISSUE_ID)
     await hass.async_block_till_done(wait_background_tasks=True)
+    assert not issue_registry.async_get_issue(DOMAIN, SUB_FAIL_ISSUE_ID)

--- a/tests/components/sonos/test_speaker.py
+++ b/tests/components/sonos/test_speaker.py
@@ -101,9 +101,6 @@ async def _media_play(hass: HomeAssistant, entity: str) -> None:
     )
 
 
-@pytest.mark.skip(
-    reason="Flaky due to Python 3.14.3 asyncio changes - see home-assistant/core#162263"
-)
 async def test_zgs_event_group_speakers(
     hass: HomeAssistant, sonos_setup_two_speakers: list[MockSoCo]
 ) -> None:

